Week2/homework/index.js

```diff
@@ -0,0 +1,101 @@
+const mysql = require('mysql');
+const prompt = require('prompt');
+const { promisify } = require('util');
+
+const connection = mysql.createConnection({
+  host: 'localhost',
+  user: 'root',
+  password: 'password',
+  database: 'new_world',
+});
+
+const execQuery = promisify(connection.query.bind(connection));
+const input = promisify(prompt.get.bind(this)); // referencing using 'this' with .bind() https://www.youtube.com/watch?v=g2WcckBB_q0
+
+async function databaseQueries() {
+  let userInputCases = '';
+
+  prompt.start();
+  try {
+    const user_choices = await input(['user_choice']);
+    userInputCases = user_choices.user_choice;
+
+    switch (userInputCases) {
+      case '1':
+        const caseOne = await input(['country']);
+        const country = caseOne.country;
+        const query1 = `SELECT name FROM city WHERE id IN (SELECT capital FROM country WHERE name = ?)`;
+
+        const result1 = await execQuery(query1, country);
+        console.table(result1);
+        break;
+
+      case '2':
+        const caseTwo = await input(['region']);
+        const region = caseTwo.region;
+        const query2 = `SELECT DISTINCT language FROM country, countryLanguage WHERE (countryCode = code) AND (region = ?)`;
+
+        const result2 = await execQuery(query2, region);
+        console.table(result2);
+        break;
+
+      case '3':
+        const caseThree = await input(['language']);
+        const language = caseThree.language;
+        const query3 = `SELECT COUNT(*) AS Cities FROM city LEFT JOIN countryLanguage ON (countryLanguage.countryCode = city.countryCode) WHERE (language = ?)`;
```

Comment:
good but I would not have used the left join but a simple (inner join). In case you have cities with no language you wouldn't count them.